game_management/game_input_handler.py

```diff
@@ -0,0 +1,593 @@
+                game_state.board.selected_unit.besieging = True
+                game_state.board.overlay.attacked_settlement.besieged = True
+                game_state.board.overlay.toggle_setl_click(None, None)
+            case _:
+                game_state.board.overlay.toggle_setl_click(None, None)
+    elif game_state.game_started and game_state.board.overlay.is_pause():
+        match game_state.board.overlay.pause_option:
+            case PauseOption.RESUME:
+                game_state.board.overlay.toggle_pause()
+            case PauseOption.SAVE:
+                save_game(game_state)
+                game_state.board.overlay.has_saved = True
+            case PauseOption.CONTROLS:
+                game_state.board.overlay.show_additional_controls = False
+                game_state.board.overlay.toggle_controls()
+            case PauseOption.QUIT:
+                game_state.game_started = False
+                game_state.on_menu = True
+                game_controller.menu.loading_game = False
+                game_controller.menu.in_game_setup = False
+                game_controller.menu.main_menu_option = MainMenuOption.NEW_GAME
+                game_controller.music_player.stop_game_music()
+                game_controller.music_player.play_menu_music()
+    elif game_state.game_started and not (
+            game_state.board.overlay.is_tutorial() or game_state.board.overlay.is_deployment() or
+            game_state.board.overlay.is_bless_notif() or
+            game_state.board.overlay.is_constr_notif() or game_state.board.overlay.is_lvl_notif() or
+            game_state.board.overlay.is_close_to_vic() or
+            game_state.board.overlay.is_investigation() or game_state.board.overlay.is_night()):
+        # If we are not in any of the above situations, end the turn.
+        if game_state.end_turn():
+            # Autosave every 10 turns.
+            if game_state.turn % 10 == 0:
+                save_game(game_state, True)
+
+            game_state.board.overlay.update_turn(game_state.turn)
+            game_state.process_heathens()
+            game_state.process_ais(game_controller.move_maker)
+
+
+def on_key_shift(game_state: GameState):
+    """
+    Handles a Shift key event in the game loop.
+    :param game_state: The current GameState object.
+    """
+    if game_state.game_started:
+        game_state.board.overlay.remove_warning_if_possible()
+        # Display the standard overlay.
+        game_state.board.overlay.toggle_standard(game_state.turn)
+
+
+def on_key_c(game_state: GameState):
+    """
+    Handles a C key event in the game loop.
+    :param game_state: The current GameState object.
+    """
+    if game_state.game_started and game_state.board.selected_settlement is not None:
+        # Pick a construction.
+        game_state.board.overlay \
+            .toggle_construction(get_available_improvements(game_state.players[0],
```

Comment:
Bit of a suspect separation of this line from the previous here, would probably be better to have the line break after the open bracket.

Reply:
[588b895](https://github.com/ChrisNeedham24/microcosm/pull/111/commits/588b895112c545f17b9883b616153d3514b540f7)